src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala

```diff
@@ -176,50 +177,71 @@ class BeamScenarioLoader(
 
   }
 
-  private[utils] def replacePlansFromPopulation(population: Population, plans: Iterable[PlanElement]): Population = {
+  private[utils] def replacePlansFromPopulation(
+    population: Population,
+    plansElements: Iterable[PlanElement]
+  ): Population = {
     logger.info("Applying plans...")
 
-    plans.foreach { planElement: PlanElement =>
-      val person = population.getPersons.get(Id.createPersonId(planElement.personId.id))
-      if (person != null) {
-        val selectedPlan = selectedPlanFromPersonOrBuildNew(person)
-        val planElementType = planElement.planElementType
-        if (planElementType.equalsIgnoreCase("leg")) {
-          planElement.legMode match {
-            case Some(mode) =>
-              PopulationUtils.createAndAddLeg(selectedPlan, mode)
-            case None =>
-              PopulationUtils.createAndAddLeg(selectedPlan, "")
-          }
-        } else if (planElementType.equalsIgnoreCase("activity")) {
-          assertActivityHasLocation(planElement)
-          val coord = if (beamScenario.beamConfig.beam.exchange.scenario.convertWgs2Utm) {
-            geo.wgs2Utm(new Coord(planElement.activityLocationX.get, planElement.activityLocationY.get))
-          } else {
-            new Coord(planElement.activityLocationX.get, planElement.activityLocationY.get)
-          }
-          val activityType = planElement.activityType.getOrElse(
-            throw new IllegalStateException(
-              s"planElement is `activity`, but `activityType` is None. planInfo: $planElement"
-            )
-          )
-          val act = PopulationUtils.createAndAddActivityFromCoord(selectedPlan, activityType, coord)
-          planElement.activityEndTime.foreach { endTime =>
-            act.setEndTime(endTime)
-          }
+    plansElements.groupBy(_.personId).foreach {
+      case (personId: PersonId, listOfElementsGroupedByPerson) =>
+        listOfElementsGroupedByPerson.groupBy(_.planIndex).foreach {
+          case (_, listOfElementsGroupedByPlan) if listOfElementsGroupedByPlan.nonEmpty =>
+            val person = population.getPersons.get(Id.createPersonId(personId.id))
+
+            val currentPlan = PopulationUtils.createPlan(person)
+            person.addPlan(currentPlan)
+
+            val personWithoutSelectedPlan = person.getSelectedPlan == null
```

Comment:
Did not we agree to explicitly show that the plan is selected by a boolean flag when we write to CSV?

Reply:
Just have seen https://github.com/LBNL-UCB-STI/beam/pull/1974, it explains